internal/cmd/server.go

```diff
@@ -98,6 +99,12 @@ func setupRunners(cfg *config.Server) error {
 	// https://github.com/envoyproxy/gateway/issues/43
 	ctx := ctrl.SetupSignalHandler()
 
+	// Setup the Extension Manager
```

Comment:
would be good to add a check here
var extMgr *pointer
```
if cfg.Extension. != nil {
   extMgr = 
```

and you can use this value nil check during cleanup

Reply:
The `extensionregistry.NewManager()` `func` now performs that check and will create an empty manager that does nothing in the case that `cfg.EnvoyGateway` or `cfg.EnvoyGateway.Extension` is `nil`. The only time the func returns `nil, error` now is when `*rest.Config for talking to a Kubernetes apiserver` cannot be created. That is an error where we quit early rather than continue and can never happen when not trying to use an extension.